### ISSUE TITLE
Vaadin pushState saves history items in browser with wrong title

### ISSUE THREAD

**Issue by Wnt** [vaadin/framework#10280]
- Vaadin Framework version 8.2.0.alpha2

When calling
`Page.getCurrent().pushState("foobar");`
and 
`Page.getCurrent().setTitle("Hello world");`
the browser (Chromium Version 62) saves the previous page in the Browser history with the wrong title. This makes navigating to a previous UI state awkward.

I suspect that this happens because `Page.getCurrent().setTitle("Hello world");` sets the title before `window.history.pushState('foobar')` is called, the old URL is stored into the browser's history with the new title.

See video demo at https://www.youtube.com/watch?v=r2IAvkXzA9w

Test application:
https://github.com/Wnt/push-state-test

Workaround:
Instead of using 
`Page.getCurrent().setTitle("Hello world");`
to set the page title, use
```
String title = "Hello world";
Page.getCurrent().getJavaScript().execute("document.title = '" + title + "';");
```
to set the title.

**Comment by Wnt** [comment]
Issue still persist in Vaadin 8.3.3 and 8.4.0.alpha1

**Comment by tsuoanttila** [comment]
Title gets updated before the Vaadin 6 style UIDLs are updated. This makes the new title be set before the history is updated. PushState needs be handled before the title, not sure how big of a refactor it would be...

**Comment by Wnt** [comment]
The issue still exist in the latest released version of Vaadin (8.5.1)
